### PR TITLE
KS-23 Shotgun

### PR DESCRIPTION
**Pull request by radar651**
Adds a new shotgun for traitor use. A KS-23 pump action shotgun. Cost 12 TC and comes loaded with 4 riot rounds. Can take any shotgun rounds available though.

**:cl: radar651 Adds KS-23 to to the game and uplink  /🆑 **

### CODE REVIEW

File: code/datums/uplink_item.dm [hunk 1]
```diff
@@ -372,6 +372,15 @@ var/list/uplink_items = list()
 	item = /obj/item/weapon/gun/projectile/automatic/pistol
 	cost = 4
 
+/datum/uplink_item/dangerous/shotgun
+	name = "KS-23 Riot control shutgun"
+	reference = "SPS"
+	desc = "A large heavy duty shotgun from the Commis. Holds 4 shotgun shells of any kind. Preloaded with rubber."
```
Comment (SamHPurp) [inline code/datums/uplink_item.dm]: "heavy duty" -> "heavy-duty", and "Commis" - > "Commies"

File: code/datums/uplink_item.dm [hunk 2]
```diff
@@ -372,6 +372,15 @@ var/list/uplink_items = list()
 	item = /obj/item/weapon/gun/projectile/automatic/pistol
 	cost = 4
 
+/datum/uplink_item/dangerous/shotgun
+	name = "KS-23 Riot control shutgun"
```
Comment (SamHPurp) [inline code/datums/uplink_item.dm]: "shutgun" -> "shotgun"

File: code/modules/projectiles/ammunition/magazines.dm [hunk 3]
```diff
@@ -446,7 +446,13 @@
...
 	max_ammo = 20
-
+	
 /obj/item/ammo_box/magazine/laser/update_icon()
 	..()
 	icon_state = "[initial(icon_state)]-[Ceiling(ammo_count(0)/20)*20]"
+	
+	
+/obj/item/ammo_box/magazine/internal/shot/KS
+	name = "KS23 shotgun internal magazine"
```
Comment (SamHPurp) [inline code/modules/projectiles/ammunition/magazines.dm]: "KS23" -> "KS-23", to keep it consistent?

**Comment by SamHPurp** [comment]
You're missing the add/del/fix/tweak symbol, also I think the Changelog lines need to be on separate lines:

`:cl: radar651
add: Adds KS-23 to to the game and uplink
/ :cl:`

**Comment by KingPhilipIII** [comment]
Shotgun for twelve TC? Is it really worth it when you could just buy an emag and switch departments into cargo?

**Comment by ZomgPonies** [comment]
Add pictures and rainbows to your description. Ok just pictures.

**Comment by Jaywings** [comment]
The KS-23 is notorious for having the biggest shell caliber ever...and being loud as fuck. Perhaps make a new type of ammo for it, and make it incredibly dangerous...as it should be.

If not, it's still a cool new little gun to toy with.

**Comment by KingPhilipIII** [comment]
Maybe make this bartender exclusive and up it's danger factor? So people know that the man serving you drinks might blow you away with a giant ass shotgun 4noraisen.

**Comment by Allfd** [comment]
This is actually an anti-aircraft barrel made into a shotgun.

It should mess you up.

**Comment by KingPhilipIII** [comment]
>In Soviet Russia, nothing go to waste, not even rejected anti-aircraft barrels.

Instead. Make a shotgun out of them.

**Comment by alexkar598** [comment]
Make it so you need to wear earmuffs or everybody including you hear a very loud sound and gets ear damage

**Comment by KingPhilipIII** [comment]
Make it cause weakness too if no earmuffs. I don't know about you but if my eardrums were just blown out I'd stop, keel over and cry a bit.

**Comment by alexkar598** [comment]
What dose weakness do?

**Comment by KingPhilipIII** [comment]
Stamina damage. It's like a less useful version of knockdowns and stuns.

**Comment by alexkar598** [comment]
Not sure it's a good idea to put a mass stun/weakness/whateveritscalled in a gun

**Comment by KingPhilipIII** [comment]
Eh. Probably not. But for 12 TC we need something with more umpf. Atleast match up to the .357

Imagine the horror of a buckshot that did 60 damage a piece.

**Comment by Fox-McCloud** [comment]
We already have 2 other ballistics for traitors, what's the point of a third?

It just becomes "pew pew gun, but more damage", which isn't particularly interesting, to be honest. 

We need less "more of the same, but better/worse for a different TC cost!" and more "this interacts with your target in ways that we haven't seen before".

**Comment by KingPhilipIII** [comment]
Unfortunantely @Fox-McCloud, there are honestly only so many ways to murder your target with uplink items. Either you beat them to death, strangle them, space them, cut them to pieces, stab them, feed them to wild life, or shoot them. And then there are various ways to do each of those. 

Then there are other ways to kill your target that just can't be done with traitor gear. No amount of clever traitoring will get you terrorspiders or xenos to feed him to, and with antagonist restrictions you can't suicide bomb them, or feed them to the engine, or set the station on fire, or other grand and destructive means.

**Comment by granodd** [comment]
Assassination certainly is one of the more difficult objectives to do in regards to getting it done and then still having TC left over to do any other difficult tasks/objectives. Would a big ol' shotgun be cool? Hell yea, I love ballistics. Does the traitor role _need_ more ballistics? No, not really. They already have the revolver for if they need to go loud and kill something, as that thing does a fair amount of damage on its own. A giant 6 gauge shotgun isn't really something they need.

**Comment by shazbot194** [comment]
While it would be un-necessary, since it shoots a 23mm shell, we could include a flare pistol (possibly for mining) that can acts as an improvised high caliber shotgun/low caliber grenade launcher and have it share ammo with the KS-23, so the KS-23 shoots ammo that is on station and not impossible to obtain while not just being another 12g shotgun.

**Comment by radar651** [comment]
I will probably add new ammo types and.. that flare gun sounds interesting in my next PR.

**Comment by Fox-McCloud** [comment]
Not updated in a while, and I strongly believe we need more unique traitor items that have unique behaviors and interact with the crew/targets in new/interesting ways, not more "point at person and click twice to kill them".

**Comment by jughu** [comment]
@fox_mccloud how about a borer extracter for 8 tc stick in targets ear and take out their borer

**Pull request closed without merging**